NavMeshComponents/Scripts/NavMeshModifierTilemap.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using NavMeshPlus.Extensions;
+using UnityEngine.Tilemaps;
+using UnityEngine.AI;
+
+//***********************************************************************************
+// Contributed by author jl-randazzo github.com/jl-randazzo
+//***********************************************************************************
+namespace NavMeshPlus.Components
+{
+    [AddComponentMenu("Navigation/Navigation Modifier Tilemap", 33)]
+    [HelpURL("https://github.com/Unity-Technologies/NavMeshComponents#documentation-draft")]
+    [RequireComponent(typeof(Tilemap))]
+    [RequireComponent(typeof(NavMeshModifier))]
+    [DisallowMultipleComponent]
+    public class NavMeshModifierTilemap : MonoBehaviour
+    {
+        [System.Serializable]
+        public struct TileModifier
+        {
+            public TileBase tile;
+            public bool overrideArea;
+            [NavMeshArea] public int area;
+        }
+
+        [SerializeField]
+        List<TileModifier> m_TileModifiers = new List<TileModifier>();
+        public IEnumerable<TileModifier> tileModifiers => m_TileModifiers;
+
+        private Dictionary<TileBase, TileModifier> m_ModifierMap;
```

Comment:
Unfortunately as far as I can tell Unity doesn't properly support serializing maps and editing them from the editor. Could use a dedicated map format for sure, but would take a while to fiddle with the editor to get that to work. I updated the code so that it caches the map OnEnable and then the editor manually updates it if any serialized properties are changed.

Reply:
Ok, cache will work. That is the main point. All other items are optional.